src/replication.c

```diff
@@ -1966,6 +2006,14 @@ handle_mgmt_conn_error(replica_t *r, int sfd, struct epoll_event *events, int ev
 	mgmtfd = r->mgmt_fd;
 	r->mgmt_fd = -1;
 	MTX_UNLOCK(&r->r_mtx);
+
+	if (r->spec->target_replica == r) {
```

Comment:
Sir do you see any correctness issue doing it at this place. I did purposely here because here we have spec->rq_mtx lock. I did not want it to be taken again. By the time we come to pthread_cond_wait, we release this lock, it means I have to take this lock again.